packages/frontend/src/contexts/Authentication.tsx

```diff
@@ -24,16 +24,12 @@ type AuthenticationProviderProps = {
 export const AuthenticationProvider = ({
   children,
 }: AuthenticationProviderProps) => {
-  const [currentUser, setCurrentUser] = useState<CurrentUser>(null)
-
-  const { loading: fetchingCurrentUser } = useQuery<{
+  const { data, loading: fetchingCurrentUser } = useQuery<{
```

Comment:
is this optimisation to drop `useState` actually related to the apollo client upgrade?

Reply:
Yes, with `useState` the upgrade breaks. Reason is explained in my description - TLDR Apollo does not update state; it looks like it was a bug / unexpected feature in earlier versions